### DISCUSSION TITLE
Zigbee Router: Feedback wanted!

### DISCUSSION THREAD

**Discussion by Nerivec** [Nerivec/silabs-firmware-builder#16] [Announcements]
Zigbee router firmware are now added for most adapters.
https://github.com/Nerivec/silabs-firmware-builder/releases/tag/v2024.6.2-update6

We need some feedback on the new features, how it performs, [issues](https://github.com/Nerivec/silabs-firmware-builder/issues/new) using it, etc.!

As a reminder, here are some details on it:

The router should automatically try to join a nearby open network every 10 seconds as long as it is not already joined (will continue indefinitely as long as it has power).

Using the `reset` button in the device's Z2M page, should allow you to enter the bootloader on the device, making flashing "from router firmware" easier (no need to open the casing, short pins, etc.). If USB, ensure you plug your device into the computer you intend to use for flashing BEFORE you press the `reset` button. Flashers should then be able to detect the launched bootloader and allow immediate flashing. _If you press this by accident, a simple unplug/short wait/replug should take your device out of the bootloader. The button is one-way, since the device should disconnect immediately after pressing it, to enter the bootloader._

More advanced use:
When plugged into a computer (or over the network for TCP adapters), you can monitor the activity on the router using a serial/TCP monitor (various terminals can do it, also the extension in VSCode, and [Ember ZLI as of 2.9.0](https://github.com/Nerivec/ember-zli/wiki/Monitor)). These logs may offer insights into misbehaving devices (e.g. during joins), from the perspective of a router, instead of the usual logs from the coordinator from Z2M.
_Note: The logs are in silabs' short-hand, which is not always easy to read, but someone should be able to help decipher, if you ask nicely!._ 😉

Devices supporting LED/button integration will have these extra features:
* The LED will turn on and off (slowly) to alert of the fact the adapter is running a new scan. Once joined, the LED will stay fixed.
* The button has 4 distinct behaviors:
  1. No network, long press (>5s): reboot into bootloader
  2. No network, short press: toggle network steering (start/stop trying to join)
  3. Joined to a network, long press (>5s): leave network
  4. Joined to a network, short press: toggle source route discovery (start/stop broadcasting MTORRs)

> [!TIP]
> Toggled statuses from button only remain effective while the adapter is powered on. Power cycling the adapter will bring them back to defaults.

> [!TIP]
> If the device doesn't appear as it should in Z2M after flashing the router firmware, and it was previously joined to Z2M, trigger a re-interview to force the information to update. Among other updated properties, `Firmware build date` should now read `2024...`.

> [!TIP]
> It is recommended to clear the NVM3 on the adapter when changing firmware type (NCP<>Router). You can find the appropriate GBLs to do so [here](https://github.com/Nerivec/silabs-firmware-recovery/tree/main?tab=readme-ov-file#nvm3-clear) or use [Ember ZLI](https://github.com/Nerivec/ember-zli/wiki/Bootloader#clear-nvm3).

### Notes on configuration for the external converter:

> [!IMPORTANT]
> Any firmware before [v2024.6.2-update4](https://github.com/Nerivec/silabs-firmware-builder/releases/tag/v2024.6.2-update4) requires the external converter, from [v2024.6.2-update4](https://github.com/Nerivec/silabs-firmware-builder/releases/tag/v2024.6.2-update4), Z2M 2.0.0 has built-in support (can remove the external converter). If you haven't upgraded to 2.0.0, keep [v2024.6.2-update3](https://github.com/Nerivec/silabs-firmware-builder/releases/tag/v2024.6.2-update3) and the external converter in place until you upgrade (as described in release notes).

> [!TIP]
> Make sure to read the [Zigbee2MQTT 2.0.0 breaking changes](https://github.com/Koenkk/zigbee2mqtt/discussions/24198) and make the appropriate changes if you are still using [v2024.6.2-update3](https://github.com/Nerivec/silabs-firmware-builder/releases/tag/v2024.6.2-update3) (with external converter) with 2.0.0. See `External converters and extensions` paragraph in previous link.

Example snippet (note the indentation):

```yaml
advanced:
  log_level: debug
  last_seen: ISO_8601_local
external_converters:
  - silabs-adapter-with-router-firmware-converter.js
```

- Make sure you use the proper folder to put the converter file in, and proper `configuration.yaml` to set it up (in HA, it's usually under the `zigbee2mqtt` folder).
- Make sure the format does not change when you copy/paste in the `configuration.yaml` (indentation is important, it should be on its own lines, and without spaces at the beginning of `external_converters:`).

**Comment by kraus-wagner** [comment]
I tried to flash your router firmware 8.0.2.0 to a sonoff dongle-e several times now. I tried the silabs web flasher from darkxst and secureCRT. No errors, everything seems to work fine. The external converter is installed too. But i am not shure.

z2m says firmware is 1.0.0 and i have no reset button?

Device type    Router
Zigbee Model    DONGLE-E_R
Zigbee Manufacturer    SONOFF
Description    Sonoff Zigbee 3.0 USB Dongle Plus (EFR32MG21) with router firmware
Support status    Supported
Firmware build date    20220515
Firmware version    1.0.0
Manufacturer    [SONOFF](https://www.zigbee2mqtt.io/supported-devices/#v=SONOFF)
Model    [ZBDongle-E](https://www.zigbee2mqtt.io/devices/ZBDongle-E.html#sonoff-zbdongle-e)

The secureCRT terminal says Sonoff v1.0.1 / Gecko Bootloader v1.12.00.

Is everything fine? How can i check if your firmware is flashed?

**Reply by Nerivec** [reply]
You flashed this file, correct?
https://github.com/Nerivec/silabs-firmware-builder/releases/download/v2024.6.2-update2/sonoff_zbdonglee_zigbee_router_8.0.2.0_115200_sw_flow.gbl

If yes, then trigger a re-interview in Z2M, if not, flash that one, and then re-interview in Z2M.

The Dongle-E will then appear like this in Z2M with this firmware:
![sonoff-router](https://github.com/user-attachments/assets/54a9ebeb-1680-4ba2-bb07-18ef02fc89c8)

**Reply by kraus-wagner** [reply]
Flashing was ok. The re-interview did it! Now everything is shown correct. Reset button is on details page!

**Reply by Belox86** [reply]
Hi @Nerivec I have the same problem. After having updated the router firmware on the dongle i still see 
```
Firmware build date    20220515
Firmware version       1.0.0
```
i tried the reconfigure button (is this what you refer as  re-interview?), and also to remove the device and re-pair it but still see the old fw version.
before the update, the command `universal-silabs-flasher --device COM5 probe` was giving me the correct version of the router fw (it was the 6. something), but now no:
```
2025-01-29 16:33:37.547 INX-LTW1901 universal_silabs_flasher.flasher INFO Probing ApplicationType.GECKO_BOOTLOADER at 115200 baud
2025-01-29 16:33:39.574 INX-LTW1901 universal_silabs_flasher.flasher INFO Probing ApplicationType.CPC at 460800 baud
2025-01-29 16:33:43.824 INX-LTW1901 universal_silabs_flasher.flasher INFO Probing ApplicationType.CPC at 115200 baud
2025-01-29 16:33:48.072 INX-LTW1901 universal_silabs_flasher.flasher INFO Probing ApplicationType.CPC at 230400 baud
2025-01-29 16:33:52.322 INX-LTW1901 universal_silabs_flasher.flasher INFO Probing ApplicationType.EZSP at 115200 baud
2025-01-29 16:33:57.326 INX-LTW1901 universal_silabs_flasher.flasher INFO Probing ApplicationType.SPINEL at 460800 baud
2025-01-29 16:34:01.598 INX-LTW1901 universal_silabs_flasher.flasher INFO Probing ApplicationType.ROUTER at 115200 baud
Error: Failed to probe running application type
```
btw, the router is operating properly, i'm just not sure whether i actually updated the fw or not. 
Im using the last version of z2m Edge without any external converters.
```
Zigbee2MQTT version                             2.0.0-dev commit: [3232969](https://github.com/Koenkk/zigbee2mqtt/commit/3232969)
Coordinator type                                EmberZNet
Coordinator revision                            8.0.2 [GA]
Coordinator IEEE Address                        0xe8e07efffe7af8ce
Frontend version                                0.9.4
zigbee-herdsman-converters version              21.26.0
zigbee-herdsman version                         3.2.4
```
Thank you

**Reply by Nerivec** [reply]
You need to use "interview", the blue button with an `i`, not "reconfigure".
Only re-interview, or remove (_not forced!_) + restart will clear the cache and force a full interview (this is meant to avoid having to re-interview as much as possible in normal operation).

The new router probing from universal silabs flasher looks to be tailored for use with the command line component, which is not included in my firmware builds (only in darkxst).
As described in first post here, you can trigger the bootloader from Z2M after the re-interview, with the toggle that's in the device's exposes. _You plug the router in the computer where you want to flash it from, then trigger the button in Z2M, it will enter bootloader, then you can flash with either Ember ZLI or Universal Silabs Flasher (will detect the gecko bootloader automatically)._

**Reply by Belox86** [reply]
oh yeah it works. Sry i didn't see the i button. Thanks again

**Comment by bangerdemon** [comment]
I flashed the file https://github.com/Nerivec/silabs-firmware-builder/releases/download/v2024.6.2-update2/sonoff_zbdonglee_zigbee_router_8.0.2.0_115200_sw_flow.gbl to a brand new Sonoff Dongle E. 

It originally showed up in z2m as unsupported. After doing other things in HA, I restarted HA, went into z2m, and the device has now disappeared!

I have tried putting it into pairing mode, but no matter what buttons are pressed, I can't get it back into pairing mode. Using the SiLabs Flasher, it can't find any firmware when I try to connect it to change back to coordinator firmware, and using SecureCRT, I can't get into the bootloader to try flashing again.

Any ideas please?

**Reply by Nerivec** [reply]
> It originally showed up in z2m as unsupported.

Did you setup the external converter? It would show up as unsupported otherwise.

While joined to a network, the Dongle-E should automatically leave the network if you press the RST button for over 5 seconds, which should automatically put it back into pairing mode (in that state, the green LED should turn on every 10sec for a short while, then repeat that pattern, it only stays fixed if it is joined to a network).
With Dongle-E, if all else fails, you should be able to enter the bootloader by pressing the BOOT button while unplugged, keep pressing, then plug it into the computer, wait a second, then release. It should then be immediately in bootloader, available for flashing with ember-zli, universal silabs flasher, SecureCRT, etc.. ember-zli can also "force reset" the Dongle-E when entering the bootloader (it'll ask), so does universal-silabs-flasher (with `--bootloader-reset sonoff` option).

_If it causes you any more troubles, try clearing the NVM3, troubles can sometimes happen with large version jumps, since that one usually still has the old v6 out-of-factory... ([ember-zli can clear the NVM3](https://github.com/Nerivec/ember-zli/wiki/Bootloader#clear-nvm3) via the bootloader menu, or you can flash the [nvm3_initfile from xsp1989 repo](https://github.com/xsp1989/zigbeeFirmware/blob/master/firmware/Zigbee3.0_Dongle-NoSigned/EZSP/nvm3_initfile.gbl), same result)._

**Reply by bangerdemon** [reply]
I have got it back into z2m, but still showing as unsupported.

It's probably me, but pretty sure I have set up the external converter correctly.
![2024-11-12](https://github.com/user-attachments/assets/bbb7fc76-5c8f-4ca5-b272-d06a71288d39)

**Reply by Nerivec** [reply]
It needs to be in the same folder as your `configuration.yaml`, and requires you add the lines mentioned in the release notes, in your `configuration.yaml`, so Z2M knows to look for it:
```yaml
external_converters:
  - silabs-adapter-with-router-firmware-converter.js
```

If it is properly included, you will see it in the first few lines in the Z2M log file:
```
z2m: Loaded external converter 'silabs-adapter-with-router-firmware-converter.js'
```

**Reply by bangerdemon** [reply]
I have added 

external_converters:
  - silabs-adapter-with-router-firmware-converter.js 
  
to the main HA configuration.yaml and to the z2m configuration.yaml. My thinking was, it will be in the correct one, but still not working.
![Screenshot_20241112-122346](https://github.com/user-attachments/assets/775845ec-7b2a-4cbb-8206-cca15d114f4a)
![Screenshot_20241112-121743](https://github.com/user-attachments/assets/8e128433-5607-47f5-8f4b-24dd2ed13446)

**Reply by kraus-wagner** [reply]
I have the config lines in the Z2M configuration.yaml
![image](https://github.com/user-attachments/assets/0a7965e7-ed34-4d2b-a57f-fb58f964f287)
And the ...js file also in the Z2M directory.
Dont forget to make a new interview.
I forgot that first.
Now everything works.

**Reply by Nerivec** [reply]
Make sure to remove those lines from the main HA configuration.yaml, it's not the right place, you don't want it to potentially conflict with anything else.

_Once it's well tested and we've had some more feedback, I'll see about including the converter directly in Z2M, it will be easier then!_

**Reply by bangerdemon** [reply]
> I have the config lines in the Z2M configuration.yaml ![image](https://private-user-images.githubusercontent.com/78031015/385296119-0a7965e7-ed34-4d2b-a57f-fb58f964f287.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE0MjI0OTksIm5iZiI6MTczMTQyMjE5OSwicGF0aCI6Ii83ODAzMTAxNS8zODUyOTYxMTktMGE3OTY1ZTctZWQzNC00ZDJiLWE1N2YtZmI1OGY5NjRmMjg3LmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExMlQxNDM2MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMjNkNGM2ZmYwYmZiYmIzMzM5MWYxNTcyNGU2ODk5NGUyMDE3Mjc1M2VkNDNiMmIyY2M5YThkNTJkYTIzNjVlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.SDdiYaUkwKPDgZYIknubbi32tguQz8zt-y5Ux6z9aa4) And the ...js file also in the Z2M directory. Dont forget to make a new interview. I forgot that first. Now everything works.

I have the config lines in z2m configuration.yaml (and I have removed it from the HA configuration.yaml), and have the .js file in the z2m folder. I have tried to reinterview the dongle, but still showing as unsupported. I have also restarted z2m numerous times!
![Screenshot_20241112-122346](https://github.com/user-attachments/assets/e99e4801-8e47-4025-8a7f-617928976a88)

**Reply by Nerivec** [reply]
I'm confused, before you said "Now everything works.", but it is still not working?
Can you post the first few lines of the log file from the last time you started Z2M?

**Reply by bangerdemon** [reply]
> I'm confused, before you said "Now everything works.", but it is still not working? Can you post the first few lines of the log file from the last time you started Z2M?

14:55:45] INFO: Preparing to start...
[14:55:45] INFO: Socat not enabled
[14:55:46] INFO: Starting Zigbee2MQTT...
Starting Zigbee2MQTT without watchdog.
[2024-11-12 14:57:20] error: 	z2m: Failed to execute routing table for 'Printer'
[2024-11-12 14:57:47] error: 	z2m: Failed to execute LQI for 'Table Lamp'

**Reply by bangerdemon** [reply]
It was another person that said that now everything works

**Reply by Nerivec** [reply]
> It was another person that said that now everything works

Argh, my bad, I read too fast!

It seems the converter is not found by Z2M. Can you make sure your `zigbee2mqtt/configuration.yaml` lines for `external_converters:` look like the one in [that screenshot](https://github.com/Nerivec/silabs-firmware-builder/discussions/16#discussioncomment-11226167). The indentation is important, there should be no space at the beginning of the line for `external_converters:`

```yaml
external_converters:
  - silabs-adapter-with-router-firmware-converter.js
```

**Reply by bangerdemon** [reply]
Now I get an error saying bad indentation!

If I uninstall z2m and then reinstall it, would I need to re pair all of my devices?

**Reply by Nerivec** [reply]
No, but you don't need to reinstall it, just fixing the indentation should be enough. Can you post the file in question?

**Reply by bangerdemon** [reply]
Tried it like this 
<img src="https://github.com/user-attachments/assets/d305ee33-70f5-4d62-8cbe-131d596afe39" height="200" title="Incorrect"/>

And like this 

![Screenshot_20241112-170751](https://github.com/user-attachments/assets/90af3c97-73a6-4b62-9f7a-616f498c169e)

**Reply by Nerivec** [reply]
The second one is correct.

PS: If needed, you can find me on Discord, you can DM me after joining the `zigbee2mqtt` server.

**Reply by bangerdemon** [reply]
I don't know if this is my problem, but if it is, I have no idea how to change it!
![Screenshot_20241112-175605](https://github.com/user-attachments/assets/dfc61a2a-86d9-4c57-9a49-22ce8a020600)

**Comment by bangerdemon** [comment]
I have sorted it!!🙏

I added the external converter into the z2m add-on configuration, and it just worked.

Thanks for all of your help, much appreciated.

**Reply by Nerivec** [reply]
Good to hear! 😉
_Seems your configuration file and the add-on configuration were out of sync, so it wasn't including it properly._

**Comment by adynis** [comment]
I need some advices: What firmware shuld I use for Home Assistant (nabucasa) yellow  ?

1. With hardware flow control or without ?
2. From where ? 

- https://github.com/darkxst/silabs-firmware-builder/tree/ember-nohw/firmware_builds/yellow
- https://github.com/darkxst/silabs-firmware-builder/tree/main/firmware_builds/yellow
- https://github.com/Nerivec/silabs-firmware-builder/releases
- https://github.com/NabuCasa/silabs-firmware/tree/main/EmberZNet
- https://github.com/NabuCasa/silabs-firmware/tree/main/EmberZNet/beta

3.  Which version?
(any version? last one ? )

**Reply by Nerivec** [reply]
It wouldn't be router firmware for the yellow, so, not quite the right place for this, but here (you'll want the `zigbee_ncp` one):
- "Stable / Official": https://github.com/NabuCasa/silabs-firmware-builder/releases
- "Edge": https://github.com/Nerivec/silabs-firmware-builder/releases

_I update the builds here fairly fast after new changes/releases, hence, you can consider them somewhat similar to the difference between Zigbee2MQTT "normal" branch, and dev/edge branch._

I keep a list with categories for `ember-zli`, which you might find useful:
https://github.com/Nerivec/ember-zli/blob/main/firmware-links.json

**Reply by thk-socal** [reply]
Just feedback on performance. I have two Zigbee networks that are using Tubes MGM24 PoE devices. One network with 170 devices and one networks with 190 devices. Just added in the SONOFF Dongle-E as additional routers using the firmware provided and last updated 4 days ago. They both joined without issue and are working perfectly as long as I remember to add the external convertor code to the Z2M installs. Performance seems stellar and networks are stable.

Everything working "out of the box" and perfectly. Thanks for such great work as always!

**Reply by adynis** [reply]
> It wouldn't be router firmware for the yellow, so, not quite the right place for this, but here (you'll want the `zigbee_ncp` one):
> 
> * "Stable / Official": https://github.com/NabuCasa/silabs-firmware-builder/releases
> * "Edge": https://github.com/Nerivec/silabs-firmware-builder/releases
> 
> _I update the builds here fairly fast after new changes/releases, hence, you can consider them somewhat similar to the difference between Zigbee2MQTT "normal" branch, and dev/edge branch._
> 
> I keep a list with categories for `ember-zli`, which you might find useful: https://github.com/Nerivec/ember-zli/blob/main/firmware-links.json

Ohh.... sorry .... my question was totally in the wrong place :(  It was about routers, not coordinators 🤦‍♂️ 
Anyhow, thanks for answering :)

**Comment by ortofan** [comment]
About to dive into v8.0.2 using 3 SLZB-6M and an SLZB-7 on two HAOS instances. 
If I am on all the latest versions, do I still need the external converter? 
And if so, what is the best way to handle individual z2m instances on the same system. Rather not park the converter into 3 z2m folder.

**Reply by Nerivec** [reply]
The external converter is required for router firmware only. For coordinator, you do not need a converter at all.

The converter is still a bit too new to include natively into Z2M, we need a bit more feedback, but hopefully it will make it into the next version (January). _Technically, the device should still do its job (routing) even if it says unsupported (no converter), but you will not have the extra features._

**Comment by ortofan** [comment]
Thanks Nerivec,
Now that I have successfully flashed and deployed 3 SLZB-06M Coordinators, I want to add 3 SLZB-06M based routers as well to 3 individual z2m instances on the same HA host. I rather not copy the same js file to each folder (I tend to forget about these things). How do I handle 3 z2m instances for the external converter? Will the next version of z2m have this built in?

**Reply by Nerivec** [reply]
Like I said above, we'll see about adding it into the January version. Until then, each Z2M instance needs the file present, so, if not a copy, a symlink, mount, etc. but that's entirely dependent on your setup. If you don't want that, you can just leave it as "unsupported" (no external converter), until the converter is added natively in Z2M; it will still do the routing, you just won't have reporting or buttons available...

**Reply by ortofan** [reply]
Funny enough, it shows supported and seem to be routing just fine and dandy :) without the converter. So go figure. One thing I noticed that on the SLZB web config page mode shows "cooridnator" after I flashed it clearly with the router. But I guess this is the same thing as the firmware date talked about previously, sort of out of your control. 

![image](https://github.com/user-attachments/assets/e7e15f51-511b-486d-8add-e689f221461e)

**Reply by Nerivec** [reply]
As yes, I forgot that the SLZB06M router has an old converter in Z2M already, so it would show supported, it just won't have the new features of the new converter (yet).
About the coordinator thing, indeed, that's probably related to the SMLight API.

**Comment by aldur11** [comment]
What's the difference between the Sonoff Dongle-E router FW here vs the 2 year old one found at https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/blob/master/Dongle-E/Router/Z3RouterUSBDonlge_EZNet6.10.3_V1.0.0.gbl ?

I flashed my Sonoff Dongle-E routers with 2 year old firmware, everything seems to work, but I'm surprised that the routers can only have 20 direct children max. Does the FW here allow for more children?

**Reply by Nerivec** [reply]
It's updated to latest firmware (8.0.2, v6 is already years old), cleaned up of unnecessary endpoints/clusters and has some extra features to make it easier to deal with (always scanning when not joined to prevent having to open the case for example, or, allowing to enter the bootloader by sending a command from Z2M directly when needing to flash, etc.). Though as mentioned, it requires the use of the provided external converter for now (not yet included into Z2M).
As for the configs, yes, they are quite different, they basically are the equivalent of the current 8.0.2 coordinator version (with a few modifications to avoid unnecessary memory allocation for a router). Values can be seen here:

https://github.com/Nerivec/silabs-firmware-builder/blob/c015a814ad9ac5fd5e8e0ad171cc27555fc32cf3/manifests/sonoff/sonoff_zbdonglee_zigbee_router.yaml#L24-L28

https://github.com/Nerivec/silabs-firmware-builder/blob/c015a814ad9ac5fd5e8e0ad171cc27555fc32cf3/manifests/sonoff/sonoff_zbdonglee_zigbee_router.yaml#L63-L83

_Note that ideally, you would want more routers (dedicated or not), each with a few devices, instead of just one with many devices, to build a proper mesh._

**Reply by aldur11** [reply]
Thanks for the reply. I completely agree that a sufficient number of routers is needed to form a good mesh. 

If I'm reading this correctly, this firmware allows a maximum of 32 direct children. Is that correct?

**Reply by Nerivec** [reply]
Indeed.
The max is technically 64, but that would be wasting memory that is otherwise allocated to improve buffering capabilities.
_A lot of non-dedicated routers (power plugs, bulbs, etc.) have 4-8, some go a little higher..._

**Comment by antst** [comment]
Aside of error I reported in #19 , it looks like router firmware is damn good ) I am running this for couple of weeks on couple of routers and everything is perfect. Even quite sensitive to everything Danfoss TRVs works as a charm behind them.
@Nerivec thanks for this one!

**Reply by kraus-wagner** [reply]
Some feedback:
i use the latest router firmware on three sonoff dongle-e. My coordinator is a SLZB06-M with latest ZB firmware too (8.0.2.0), and ESP firmware 2.6.8dev21. Everything works like a charm.
Updating of the sonoffs is now very easy. Plug it into the pc, push the reset button on the devices detail page in zigbee2mqtt, device enters bootloader and is ready for flashing. Perfect! Thank you!

**Comment by snippem** [comment]
@Nerivec a question maybe there is a abvious solution (i hope so)
I own several silabs sticks and want to use my dongle-e sticks as routers. Now i have 2 network as production 1 running Z-stack and one Ember. On the Z-stack network the dongle-e router works with your software. But when i want the dongle-e router on the ember network it does not get recognized. In my opinion this is deu to the fact both mac-adresses are the same on the router i have : 0xe0798dfffea7bd73
and the coordinator has the same adres: 0xe0798dfffea7bd73. Is there a way to change the mac-adres of the routers? As the second dongle-e is not recognized on the Z-stack network maybe deu to the fact that the mac-adress is the same to.

**Comment by bangerdemon** [comment]
Hi Nerivec. Since the latest z2m update, the Sonoff dongle e's have become unsupported again. The external converter doesn't seem to work anymore. Any advice please?

**Reply by Nerivec** [reply]
Update4 provides firmware that should work without the need for an external converter (you can just remove it).
Make sure to restart Z2M and re-interview.

**Reply by bangerdemon** [reply]
Have I got to take them apart again to re-interview? 
I have restarted z2m numerous times after removing the external converter and pressed the "I" button numerous times, but nothing has happened.

**Reply by Nerivec** [reply]
What problem are you seeing exactly?
_Any firmware before `update4` requires the external converter, from `update4`, Z2M 2.0.0 has built-in support. If you haven't upgraded to 2.0.0, keep update3 and the external converter in place until you upgrade (as described in release notes)._

**Reply by bangerdemon** [reply]
I have upgraded z2m to 2.0.0.2

**Reply by bangerdemon** [reply]
The Sonoff dongle e's are showing as unsupported. The lqi on theM hasn't changed at all today.

**Reply by bangerdemon** [reply]
![Screenshot_20250107-174405](https://github.com/user-attachments/assets/ab0ba925-ef94-4f10-ae07-16bd1fa2cf19)
![Screenshot_20250107-174350](https://github.com/user-attachments/assets/b08ec1b5-6a7c-4ac8-b5ce-c4e69aed2344)

**Reply by Nerivec** [reply]
Assuming it worked when you did the re-interview (no error?) you are using an older firmware, that means you need to update the router firmware on the adapter.
As mentioned, for use with Zigbee2MQTT 2.0.0 without external converter, you need `update4` minimum:
https://github.com/Nerivec/silabs-firmware-builder/releases/tag/v2024.6.2-update4

---

_It was not possible to add built-in support in Z2M without clashing with "official manufacturer firmware", so I had to differentiate the builds using a unique property, which is required by Z2M 2.x.x to recognize them. Older firmware do not have that property, hence, not recognized._

**Comment by quodo** [comment]
Thanks @Nerivec, updating my firmware to 8.0.2 solved a lot of `Failed to send with status=BUSY errors` after switching from ezsp to ember.

[This comment](https://github.com/Koenkk/zigbee2mqtt/discussions/23224#discussioncomment-10412608) about being able to revert to v7 by @chris-1243 made me dare to try.

Initial pointer with the same issue from @corporategoth at https://github.com/Koenkk/zigbee2mqtt/issues/24249.

**System**

- Home Assistant v2025.1.2
- Zigbee2MQTT v2.0.0-2, ember, "Home Assistant legacy action sensors" unchecked
- Sonoff Zigbee 3.0 USB Dongle E
- Old Firmware Zigbee dongle: `ncp-uart-hw-v7.4.5.0-zbdonglee-115200.gbl` from https://github.com/darkxst/silabs-firmware-builder/tree/main/firmware_builds/zbdonglee.

**Example error**
`z2m: Publish 'set' 'state' to 'Lampje TV dressoir' failed: 'Error: Command 2 genOnOff.on({}) failed (~x~> [ZCL GROUP groupId=2] Failed to send with status=BUSY.)'`

**Updating the firmware** of my Sonoff Zigbee 3.0 USB Dongle E to 
`sonoff_zbdonglee_zigbee_ncp_8.0.2.0_115200_sw_flow.gbl` found here https://github.com/Nerivec/silabs-firmware-builder/releases (v2024.6.2-update5) using the Firmware Flasher found here https://darkxst.github.io/silabs-firmware-builder/ solves my problem with `Failed to send with status=BUSY errors`.

**Result**
Happy husbands and kids who can use the Ikea styrbar remote control again for several lights  :-D

**Comment by chris-1243** [comment]
I flashed a Dongle-E with your latest zigbee router build and so far so good. For me, there is no change in my network compared to the old stock router firmware (6.70.xxx).

**EDIT** This firmware is running fine since 14 days. ZGP devices are well routed as well and not a single failure.👍

**Comment by chs72** [comment]


**Reply by Nerivec** [reply]
You would have to enable the CLI in the router build, and manage it with it.
https://docs.silabs.com/zigbee/latest/zigbee-app-framework-dev-guide-sdk-7x/11-the-command-line-interface-cli

**Reply by chs72** [reply]
> You would have to enable the CLI in the router build, and manage it with it. https://docs.silabs.com/zigbee/latest/zigbee-app-framework-dev-guide-sdk-7x/11-the-command-line-interface-cli

How can I enable the CLI in your project

**Reply by chs72** [reply]
[rexense_rex3b215_zigbee_router.yaml](https://github.com/chs72/silabs-firmware-builder/blob/sisdk-2024.6.2/manifests/rexense/rexense_rex3b215_zigbee_router.yaml)
Here is my manifest file.

**Comment by mamrai1** [comment]
Wow! Sonoff dongle-e with 2025.6.2_Nerivec (8.2.2) router running great! Zero problems with especially ikea parasoll contact sensors

**Reply by Nerivec** [reply]
Did you notice a difference with the previous firmware you had? Or did you have the old Sonoff 6.10.3 (that one is from 2021, so probably more likely to cause issues nowadays)?

**Comment by mamrai1** [comment]
No, I had 8.2.1. Zero problems with that also

**Comment by Asmir1975** [comment]
Updated Sonoff Dongle-E to v2026.6.0-pre1 from 8.0.2. Working well, minor UI bug, no issues 👍

<img width="784" height="628" alt="Sonoff Dongle E missmatch" src="https://github.com/user-attachments/assets/4f596f5f-f26b-49f3-9267-5b56b2ff0c45" />

**Comment by prucho2** [comment]
Hello .

Is router firmware version 8.2.2.0 the last one that works in Z2M? 
For me, all subsequent versions (v2025.12.2, v2025.12.3, v2026.6.0, v2026.6.1) only function when "permit join" is enabled. If I turn it off, they stop communicating entirely. Once I re-enable "permit join," they start communicating again every 20–30 seconds. 
Clearing NVM3 doesn't help. 
I’ve tested this on the Sonoff Dongle-E, EasyIoT ZB-GW04-1v2, and Sonoff Dongle-P (MG24). 
With all these newer firmware versions, the power icon shows a "?" and the devices fail to complete automatic configuration after joining:

z2m: Configuring '0x1cc0760001eba28c'

z2m: Failed to configure '0x1cc0760001eba28c', attempt 1 (Error: Bind 0x1cc0760001eba28c/1 genBasic from '0x04e3e5fffe80273d/1' failed (Status 'NOT_AUTHORIZED') at Endpoint.bind (/app/node_modules/.pnpm/zigbee-herdsman@10.9.5_supports-color@7.2.0/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:734:23) at async setupAttributes (/app/node_modules/.pnpm/zigbee-herdsman-converters@26.111.0_supports-color@7.2.0/node_modules/zigbee-herdsman-converters/src/lib/modernExtend.ts:204:13) at async configure (/app/node_modules/.pnpm/zigbee-herdsman-converters@26.111.0_supports-color@7.2.0/node_modules/zigbee-herdsman-converters/src/lib/modernExtend.ts:268:17) at async Object.configure (/app/node_modules/.pnpm/zigbee-herdsman-converters@26.111.0_supports-color@7.2.0/node_modules/zigbee-herdsman-converters/src/index.ts:371:21) at async Configure.configure (/app/lib/extension/configure.ts:135:13) at async EventEmitter.wrappedCallback (/app/lib/eventBus.ts:253:17))

In my Zigbee network, I use an SMlight SLZB-07MG24 coordinator—always running the latest test firmware ncp_2026.6.1_9.1.1_460800—along with two Sonoff Dongle-P (MG24) routers. However, I am forced to use version v8.2.2 on the routers because it is the last version that continues to function after the "permit join" period ends. Could I be doing something wrong? I’ve already tried various combinations, such as removing the routers from the network and so on.

Regards
prucho

**Reply by Nerivec** [reply]
Moved to https://github.com/Nerivec/silabs-firmware-builder/issues/126
